src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNICommon.cs

```diff
@@ -44,6 +44,46 @@ internal class SNISMUXHeader
         public uint length;
         public uint sequenceNumber;
         public uint highwater;
+
+        public void Read(ReadOnlySpan<byte> bytes)
+        {
+            SMID = bytes[0];
+            flags = bytes[1];
+            sessionId = BitConverter.ToUInt16(bytes.Slice(2,2));
```

Comment:
May I suggest slicing into a temp variable and use that one? Then you don't need to manually track the start offset, makes it easier to read.

Reply:
The only way I can think to do that seems less clear to me, can you give an example?

Reply:
Sorry. My bad, I've read your code wrong, my suggestion doesn't work.

Reply:
Thanks for looking it over 😁